### ISSUE TITLE
SSR?

### ISSUE THREAD

**Issue by drejohnson** [apollographql/reason-apollo#12]
Are there plans to get this to work for server-side rendering?

**Comment by Gregoirevda** [comment]
Could you provide some more details of what isn't working or features missing?

**Comment by drejohnson** [comment]
Was getting `Error: fetch is not found globally and no fetcher passed, to fix pass a fetch for
[1]       your environment like https://www.npmjs.com/package/node-fetch.` but fixed it by adding `[%bs.raw "require('isomorphic-unfetch')"];`

But I thinking more about how to do rehydration on the client like the [docs](https://www.apollographql.com/docs/react/recipes/server-side-rendering.html#store-rehydration) suggest.

**Comment by Gregoirevda** [comment]
Part of that should be fixed now: 
- Being able to set ssr to true: https://github.com/apollographql/reason-apollo/blob/master/src/ReasonApollo.re#L5

- Being able to provide a fetch client: https://github.com/apollographql/reason-apollo/blob/master/src/ApolloLinks.re#L24

**Comment by Gregoirevda** [comment]
Being able to restore the in memory cache: https://github.com/apollographql/reason-apollo/pull/25

**Comment by drejohnson** [comment]
Yep, this is pretty much resolved. Could add bindings for the `getDataFromTree` and the `renderDataFromString` functions for server, as well as the `extract()` method to be piped onto the initialized client e.g. `client.extract()`.

**Comment by kgoggin** [comment]
Anyone have an example of this in the wild? I'm curious about how you'd set up a different instance of the client for server use and then have that client used when doing SSR in place of the one defined for the client. Thanks!

**Comment by BeerPoldet** [comment]
It would be great for an example. cheers!

**Comment by rickiesmooth** [comment]
yes I was also looking for an example of providing a fetch client. I tried 
```
  ApolloLinks.createHttpLink(
    ~fetch=Fetch,
  );
```
But that didn't work.

**Comment by romanschejbal** [comment]
Queries don't seem to execute on the server, am I missing something?

**Comment by EnriqueVidal** [comment]
@romanschejbal  queroes execute for me, could be a network issue? like host unreachable? what is the error thrown?
